### DISCUSSION TITLE
WinForms: Using WhenAnyValue with BindingList item count change.

### DISCUSSION THREAD

**Discussion by vector-man** [reactiveui/ReactiveUI#2826] [Q&A]
I need to create a command with a condition in the view model for WhenAnyValue that will work with a BindingList's Item Count change (the BindingList property in the view model is bound to a control on the form.)

When the item count is 0, I need the command to not work (reactive UI isn't seeing the change for me.)

Any help is appreciated.

**Comment by glennawatson** [comment]
DynamicData can probably help since it has native understanding of BindingList's.

The issue is BindingList's count property is not implementing INotifyPropertyChanged.

but something like

`myBindingList.ToObservableChangeSet().ToCollection().Select(x => x.Count != 0)`

**Reply by vector-man** [reply]
Not sure of the syntax for that. This is not working:
`  ShowProcessing = ReactiveCommand.Create(() => {
   ...             
}, this.WhenAnyValue(x => x.Identifying, y => y.Updating, z => z.Targets, (x, y, z) => !(x || y) && z.Any()), RxApp.MainThreadScheduler);
`
Targets is the BindingList.

How would I use your code in that command? Would I bind the result to a variable in the ViewModel then check that in WhenAnyValue?
